src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -54,6 +60,21 @@ export function TopNavMenu(props: TopNavMenuProps) {
     });
   }
 
+  function renderMenu() {
+    if (!showTopNavMenu) return;
```

Comment:
The new `showTopNavMenu` prop avoids rendering the `kbnTopNavMenu` element when it's not required, and allows hiding the menu even if menu items are passed. This could also be done by conditionally deciding whether to pass no `config`, however it would still render the empty element before this change and I didn't want to affect the existing behaviour in case something else is relying on the additional padding the element introduces.

Reply:
I think it's ok to remove it.